src/cli/cluster/cluster_manager.js

```diff
@@ -29,15 +29,14 @@ import { REPO_ROOT } from '@kbn/dev-utils';
 import Log from '../log';
 import Worker from './worker';
 import { Config } from '../../legacy/server/config/config';
-import { transformDeprecations } from '../../legacy/server/config/transform_deprecations';
 
 process.env.kbnWorkerType = 'managr';
 
 export default class ClusterManager {
   static create(opts, settings = {}, basePathProxy) {
     return new ClusterManager(
       opts,
-      Config.withDefaultSchema(transformDeprecations(settings)),
+      Config.withDefaultSchema(settings),
```

Comment:
Not sure if it works as expected. As I understand `settings` is expected to be POJO here. But we pass `LegacyConfig` https://github.com/elastic/kibana/pull/52251/files#diff-748376502c51ed8bfc271f4a6c4c26d3R251
Shouldn't we wrap in `getLegacyRawConfig` or pass `this.settings` in `legacy_service` ?

Reply:
I think you are right. However the PR did not change the arguments we were invoking ClusterManager.create with. So I guess this has always been broken?

The fix is (well, should be) easy
```typescript
require('../../../cli/cluster/cluster_manager').create(
      this.coreContext.env.cliArgs,
--    config,
++    config.get(),
      await basePathProxy$.toPromise()
    );
```
However the legacy service tests are a mess, and are not properly mocking with correct object types (the config in test is actually a `Record` and not a `LegacyConfig` ). Already had to fix that in https://github.com/elastic/kibana/pull/52060, so would be easier to fix in a separate PR once it lands.

Will open a ticket for that if it's alright with you

Reply:
https://github.com/elastic/kibana/issues/52860 created

Reply:
yeah, not a blocker for the current work.